### PR TITLE
Put all params for commit into a single CommitParam struct

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @toregge PR
This is the non semantic changes from #16659
The semantic changes will come in a separate PR

### CODE REVIEW

File: searchcore/src/tests/proton/attribute/attributeflush_test.cpp [hunk 1]
```diff
@@ -161,11 +161,11 @@ UpdaterTask::run()
             }
             ia.update(j, i);
         }
-        ia.commit(i-1, i); // save i as last sync token
+        ia.commit(CommitParam(i-1, i)); // save i as last sync token
```
Comment (toregge) [inline searchcore/src/tests/proton/attribute/attributeflush_test.cpp:164]: i will be interpreted as bool here. Consider adding a third argument `false` to ensure i is considered to be a serial number.

**Review by toregge — CHANGES_REQUESTED** [review]

File: searchcore/src/tests/proton/attribute/attributeflush_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/attribute/attributeflush_test.cpp:164]: Ouch, then we have some very bad signatures here. Must fix.

Comment (baldersheim) [reply searchcore/src/tests/proton/attribute/attributeflush_test.cpp]: Fixed, PTAL

**Review by toregge — APPROVED** [review]

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**